services/user/userService.ts

```diff
@@ -0,0 +1,46 @@
+import axios, { AxiosInstance, AxiosResponse } from 'axios';
+import { assert, ResponseEnvelope, unwrapOrThrowError } from '../utils';
+import { GetUserResponse } from './types';
+import store from '../../store/App.store';
+
+export default class UserServiceClient {
+  private _axios: AxiosInstance;
+  private _baseURL: string;
+
+  constructor(serviceUrl?: string) {
+    const baseURL =
+      serviceUrl || `http://${process.env.BACKEND_URL}/api/v1/users`;
+    this._baseURL = baseURL;
+    assert(baseURL);
+    this._axios = axios.create({
+      baseURL: this._baseURL,
+      withCredentials: true,
+    });
+
+    // add cookies to every request
```

Comment:
Interceptors are cool! We use this to configure every request to add a customizable config -- for now it just has the headers that I **despise**